File: app_web_net/angular_frontendWeb/src/app/interfaces/materiaprima.ts
export interface Proveedor {
  idProveedor: number;
  nombre: string;
}

export interface MateriaPrimaResumen {
  idMateriaPrima: number;
  nombre: string;
  stock: number;
  stockMinimo: number;
}

export interface MateriaPrima {
  idMateriaPrima: number;
  nombre: string;
  descripcion?: string | null;
  costoUnitario: number;
  precioPromedio?: number | null;
  stock: number;
  stockMinimo: number;
  idProveedor: number;
  proveedor?: Proveedor | null;
}

// Lo que se envía en POST/PUT api/materias-primas
export interface MateriaPrimaCreateDto {
  nombre: string;
  descripcion?: string | null;
  costoUnitario: number;
  stock: number;
  stockMinimo: number;
  idProveedor: number;
}

File: app_web_net/backend_net/Models/MateriaPrima.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CangureraInteligente.Models;

[Table("MateriaPrima", Schema = "Operativo")]
public class MateriaPrima
{
    [Key]
    public int IdMateriaPrima { get; set; }

    [Required]
    [StringLength(100)]
    public string Nombre { get; set; } = string.Empty;

    public string? Descripcion { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal CostoUnitario { get; set; }

    public int Stock { get; set; }

    public int StockMinimo { get; set; }

    [Required]
    public int IdProveedor { get; set; }

    [ForeignKey(nameof(IdProveedor))]
    public Proveedor? Proveedor { get; set; }
   public ICollection<ProductoMateriaPrima> Productos { get; set; }
}
